plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/security/KafkaSecurityConfig.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Config("kafka.ssl.keystore.location")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_LOCATION_DOC)
+    public KafkaSecurityConfig setKeystoreLocation(String keystoreLocation)
+    {
+        this.keystoreLocation = keystoreLocation;
+        return this;
+    }
+
+    public Optional<String> getKeystorePassword()
+    {
+        return Optional.ofNullable(keystorePassword);
+    }
+
+    @Config("kafka.ssl.keystore.password")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public KafkaSecurityConfig setKeystorePassword(String keystorePassword)
+    {
+        this.keystorePassword = keystorePassword;
+        return this;
+    }
+
+    public Optional<KafkaKeystoreTruststoreType> getKeystoreType()
+    {
+        return Optional.ofNullable(keystoreType);
+    }
+
+    @Config("kafka.ssl.keystore.type")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_TYPE_DOC)
+    public KafkaSecurityConfig setKeystoreType(String keystoreType)
+    {
+        this.keystoreType = KafkaKeystoreTruststoreType.fromString(keystoreType);
+        return this;
+    }
+
+    public Optional<String> getTruststoreLocation()
+    {
+        return Optional.ofNullable(truststoreLocation);
+    }
+
+    @Config("kafka.ssl.truststore.location")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_LOCATION_DOC)
+    public KafkaSecurityConfig setTruststoreLocation(String truststoreLocation)
+    {
+        this.truststoreLocation = truststoreLocation;
+        return this;
+    }
+
+    public Optional<String> getTruststorePassword()
+    {
+        return Optional.ofNullable(truststorePassword);
+    }
+
+    @Config("kafka.ssl.truststore.password")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public KafkaSecurityConfig setTruststorePassword(String truststorePassword)
+    {
+        this.truststorePassword = truststorePassword;
+        return this;
+    }
+
+    public Optional<KafkaKeystoreTruststoreType> getTruststoreType()
+    {
+        return Optional.ofNullable(truststoreType);
+    }
+
+    @Config("kafka.ssl.truststore.type")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_TYPE_DOC)
+    public KafkaSecurityConfig setTruststoreType(String truststoreType)
+    {
+        this.truststoreType = KafkaKeystoreTruststoreType.fromString(truststoreType);
+        return this;
+    }
+
+    public Optional<KafkaEndpointIdentificationAlgorithm> getEndpointIdentificationAlgorithm()
+    {
+        return Optional.ofNullable(endpointIdentificationAlgorithm);
+    }
+
+    @Config("kafka.ssl.endpoint-identification-algorithm")
+    @ConfigDescription(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DOC)
+    public KafkaSecurityConfig setEndpointIdentificationAlgorithm(String endpointIdentificationAlgorithm)
+    {
+        this.endpointIdentificationAlgorithm = KafkaEndpointIdentificationAlgorithm.fromString(endpointIdentificationAlgorithm);
```

Comment:
do not call `fromString` explicitly here.
Instead change setter argument from `String endpointIdentificationAlgorithm` to `KafkaEndpointIdentificationAlgorithm endpointIdentificationAlgorithm`.

Update other similar places too.